src/main/java/org/rumbledb/expressions/module/FunctionDeclaration.java

```diff
@@ -48,6 +50,10 @@ public Expression getExpression() {
         return this.functionExpression;
     }
 
+    public FunctionIdentifier getFunctionIdentifier() {
+        return new FunctionIdentifier(this.functionExpression.getName(), this.functionExpression.getParams().size());
```

Comment:
How about creating this identifier in the constructor and storing it in a field?  This would avoid the repeated object creation at each call.

Reply:
done, in the InlineFunctionExpression class.